test/e2e/gc_e2e_test.go

```diff
@@ -34,159 +33,268 @@ var _ = Describe("Garbage collector", func() {
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "dependent",
-			},
-			Data: map[string]string{},
-		}
-
-		// add owners
-		ownerutil.AddOwner(dependent, fetchedA, true, false)
-		ownerutil.AddOwner(dependent, fetchedB, true, false)
-
-		// create dependent
-		_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Create(context.TODO(), dependent, metav1.CreateOptions{})
-		require.NoError(GinkgoT(), err, "dependent could not be created")
-
-		// delete ownerA in the foreground (to ensure any "blocking" dependents are deleted before ownerA)
-		propagation := metav1.DeletionPropagation("Foreground")
-		options := metav1.DeleteOptions{PropagationPolicy: &propagation}
-		err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(context.TODO(), fetchedA.GetName(), options)
-		require.NoError(GinkgoT(), err)
-
-		// wait for deletion of ownerA
-		require.NoError(GinkgoT(), waitForDelete(func() error {
-			_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), ownerA.GetName(), metav1.GetOptions{})
-			return err
-		}))
-
-		// check for dependent (should still exist since it still has one owner present)
-		_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(context.TODO(), dependent.GetName(), metav1.GetOptions{})
-		require.NoError(GinkgoT(), err, "dependent deleted after one owner was deleted")
-		GinkgoT().Log("dependent still exists after one owner was deleted")
-
-		// delete ownerB in the foreground (to ensure any "blocking" dependents are deleted before ownerB)
-		err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(context.TODO(), fetchedB.GetName(), options)
-		require.NoError(GinkgoT(), err)
-
-		// wait for deletion of ownerB
-		require.NoError(GinkgoT(), waitForDelete(func() error {
-			_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), ownerB.GetName(), metav1.GetOptions{})
-			return err
-		}))
-
-		// check for dependent (should be deleted since last blocking owner was deleted)
-		_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(context.TODO(), dependent.GetName(), metav1.GetOptions{})
-		require.Error(GinkgoT(), err)
-		require.True(GinkgoT(), k8serrors.IsNotFound(err))
-		GinkgoT().Log("dependent successfully garbage collected after both owners were deleted")
+	// TestOwnerReferenceGCBehavior runs a simple check on OwnerReference behavior to ensure
+	// a resource with multiple OwnerReferences will not be garbage collected when one of its
+	// owners has been deleted.
+	// Test Case:
+	//				CSV-A     CSV-B                        CSV-B
+	//				   \      /      --Delete CSV-A-->       |
+	//				   ConfigMap						 ConfigMap
+	Describe("Resources with multiple OwnerReferences", func() {
+
+		Context("Given a dependent resource associated with multiple owners", func() {
```

Comment:
These contexts seem redundant to me. Can they be condensed into a single container node? I also have the same feedback about the context description matching the state of the cluster -- the resources should be created here if the context description says they are given.